enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,512 @@
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
+for HyperShift)
+
+
+
+## Proposal
+
+**Cloud Credential Operator (CCO) changes**: Adds a token-aware mode, while nominally set in "Manual" mode, CCO will
+look for and process CredentialRequests referenced in Operator CRs when:
+
+```bash
+# infrastructure platform is AWS
+$ oc get infrastructure cluster -o jsonpath --template '{ .status.platform }'
+AWS
+
+# credentialsMode is "Manual"
+$ oc get cloudcredential cluster -o jsonpath --template '{ .spec.credentialsMode }'
+Manual
+
+# serviceAccountIssuer is non empty
+$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'
+abutcher-oidc.s3.us-east-1.amazonaws.com
+```
+
+This will operate in one of two ways depending on which other
+parts of this EP for other components are adopted: specifically HyperShift/Rosa adding CCO with these token-aware
+changes or continuing with the current support for the pod identity webhook.
+
+CCO's token-aware mode will work by adding a "role-arn"-type field on operator added CredentialsRequest objects.
+This is a new API field on the CredentialsRequest spec. When CCO acquires a CredentialsRequest it will create a Secret 
+(as today mostly) with new functionality: Adding to the Secret a path to the projected ServiceAccount token 
+(static for OCP) and a roleARN (from the new field on the CredentialsRequest).
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for import of additional, well-known, bundle annotations which will result 
+in ENV variables on the Subscription object. Show operator is enabled for Token-based use based on Console determining 
+cluster is running in token-based authentication enabled cluster by reading the `.spec.serviceAccountIssuer` from the 
+Authentication CR, `.spec.platformSpec.type` from the Infrastructure CR,`.spec.credentialsMode` from the 
+CloudCredentials CR  AND bundle has token-based auth enabled annotation. This info should be written to a new field on 
+Subscription objects. The Subscription fields will allow UX for the information needed by CCO or webhook, for input of 
+the cloud credentials. Generate a manual ack for operator upgrade when cloud resources are involved. This can be 
+determined by parsing the Subscription objects. Ensures admin signs off that cloud resources are provisioned and working
+as intended before an operator upgrade.
```

Comment:
> new ... Subscription fields

Is this proposing a change to the OLM Subscription API?

> Generate a manual ack for operator upgrade when cloud resources are involved.

It isn't clear to me what this means. If the Subscription has `spec.installPlanApproval: Automatic`, but an upgrade requires cloud API permission changes, it seems like we would be violating user intent by injecting a manual ack.

But this leads me to a boarder question: at what point of the OLM install/upgrade progression would this manual ack be injected, and how would it be surfaced to a user? Would OLM be expected to check the permissions so that the ack is only required if permissions need to change, or are we expecting users to manually ack every single install/upgrade involving an STS-supporting bundle deployed on an STS-enabled cluster?